rpc/storage.go

```diff
@@ -0,0 +1,278 @@
+package rpc
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie"
+	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/jsonrpc"
+)
+
+/****************************************************
+		Storage Handlers
+*****************************************************/
+
+// StorageAt gets the value of the storage at the given address and key.
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L110
+func (h *Handler) StorageAt(address, key felt.Felt, id BlockID) (*felt.Felt, *jsonrpc.Error) {
+	stateReader, stateCloser, rpcErr := h.stateByBlockID(&id)
+	if rpcErr != nil {
+		return nil, rpcErr
+	}
+	defer h.callAndLogErr(stateCloser, "Error closing state reader in getStorageAt")
+
+	value, err := stateReader.ContractStorage(&address, &key)
+	if err != nil {
+		return nil, ErrContractNotFound
+	}
+
+	return value, nil
+}
+
+// StorageProof returns the merkle paths in one of the state tries: global state, classes, individual contract
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/647caa00c0223e1daab1b2f3acc4e613ba2138aa/api/starknet_api_openrpc.json#L910
+func (h *Handler) StorageProof(id BlockID, classes, contracts []felt.Felt, storageKeys []StorageKeys) (*StorageProofResult, *jsonrpc.Error) {
+	stateReader, stateCloser, err := h.bcReader.HeadState()
+	if err != nil {
+		return nil, ErrInternal.CloneWithData(err)
+	}
+	defer h.callAndLogErr(stateCloser, "Error closing state reader in getStorageProof")
+
+	head, err := h.bcReader.Head()
+	if err != nil {
+		return nil, ErrInternal.CloneWithData(err)
+	}
+
+	if !(id.Latest || head.Number == id.Number) {
```

Comment:
Should we also allow the users to specify the blockhash in the BlockID?

Reply:
Alos, from the users perspective, this method will only seem "stable" if they set blockID="latest", as they are unlikely to know the block number or block hash that the juno node is on, right?

Reply:
For now - yes. 
Maybe I'll stick to "latest" as an only available option - otherwise it will work very strange from the users perspective 🤔